linkerd/app/gateway/src/make.rs

```diff
@@ -0,0 +1,110 @@
+use super::gateway::Gateway;
+use futures::{future, Future, Poll};
+use linkerd2_app_core::proxy::api_resolve::Metadata;
+use linkerd2_app_core::{dns, transport::tls, Error, NameAddr};
+use linkerd2_app_inbound::endpoint as inbound;
+use linkerd2_app_outbound::endpoint as outbound;
+use std::net::IpAddr;
+use std::sync::Arc;
+
+#[derive(Clone, Debug)]
+pub(crate) struct GatewayMake<R, O> {
```

Comment:
very nitpicky, feel free to ignore: it seems like we normally call `MakeService` impls `MakeFoo` rather than `FooMake`? should this be made consistent?